src/app/recommendations.ts

```diff
@@ -296,4 +296,187 @@ export const RECOMMENDATIONS: Step[] = [
+    action: "Remove all imports of `ServerTransferStateModule` from your application. It is no longer needed."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Basic,
+    step: "18.0.0. Update `Route.redirectTo` to accept functions",
+    action: "Update all `redirectTo` uses on your `Route` objects to accept functions. If you were previously using a string, you will need to wrap it in a function."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Basic,
+    step: "18.0.0. Use `afterNextRender` after change detection",
+    action: "Change code which relies on change detection timing to always use `afterNextRender` instead of `setTimeout`."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Basic,
+    step: "18.0.0: Adjust guards to return `RedirectCommand`",
+    action: "Adjust any guards that previously returned only `boolean` or `UrlTree` values to now return `RedirectCommand`."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Basic,
+    step: "18.0.0: Mark root `OnPush` views dirty",
+    action: "For any root components using `OnPush` change detection, ensure they are properly marked dirty to enable host binding updates."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Advanced,
+    step: "18.0.0-Refresh-Newly-Created-Views",
+    action: "Be aware that newly created views or views marked for check and reattached during change detection are now guaranteed to be refreshed in that same change detection cycle."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Medium,
+    step: "18.0.0-Ensure-Change-Detection-Completes-If-Marked",
+    action: "Ensure that change detection finishes eventually, or avoid marking views for check multiple times. Alternatively, you'd get an error `NG0103`"
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Medium,
+    step: "18.0.0: Ensure correct test setup",
+    action: "After aligning the semantics of `ComponentFixture.whenStable` and `ApplicationRef.isStable`, your tests may wait for unfinished navigation and requests. You can identify open requests by adding `HttpTestingController.verify()` before `await fixture.whenStable`. To workaround the new behavior you can remove the `whenStable`, instead wait for another condition, or ensure `HttpTestingController` mocks responses for all requests.",
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Basic,
+    step: "18.0.0. Update test to accommodate fixture auto detection",
+    action: "You may experience tests failures if you have tests that rely on change detection execution order when using `ComponentFixture.autoDetect` because it now executes change detection for fixtures within `ApplicationRef.tick`."
```

Comment:
```suggestion
    level: ApplicationComplexity.Advanced,
    step: "18.0.0. `ComponentFixture.autoDetect` behavior more closely matches Application behavior",
    action: "You may experience tests failures if you have tests that rely on change detection execution order when using `ComponentFixture.autoDetect` because it now executes change detection for fixtures within `ApplicationRef.tick`. For example, this will cause test fixture to refresh before any dialogs that it creates whereas this may have been the other way around in the past."
```